ma/maDBG.cc

```diff
@@ -42,7 +43,28 @@ void writeMesh(ma::Mesh* m,
     ss << prefix;
   std::string tmp = ss.str();
   const char* fileName = tmp.c_str();
-  apf::writeVtkFiles(fileName, m);
+  apf::writeVtkFiles(fileName, m, dim);
+}
+
+void addClassification(ma::Adapt* a,
+    const char* fieldName)
+{
+  ma::Mesh* m = a->mesh;
+  apf::Field* field;
+  field = m->findField(fieldName);
+  if (field)
+    apf::destroyField(field);
+
+  field = apf::createFieldOn(m, fieldName, apf::SCALAR);
+  ma::Entity* ent;
+  ma::Iterator* it;
+  it = m->begin(0);
+  while ( (ent = m->iterate(it)) ){
+    ma::Model* g = m->toModel(ent);
+    double modelDimension = (double)m->getModelType(g);
+    apf::setComponents(field, ent, 0, &modelDimension);
+  }
```

Comment:
there are get/set APIs for classification in mds, use them here instead of this. Also at what point in the workflow are you finding the mesh does not have classifn info  and need to query model? if need to write anew , mds is a better place for it